source/visionEnhancementProviders/screenCurtain.py

```diff
@@ -300,6 +300,61 @@ def confirmInitWithUser(self) -> bool:
 			return res == wx.YES
 
 
+def isScreenFullyBlack() -> bool:
+	"""
+	Takes a screen capture of all displays, through wxPython, which uses the winGDI API and checks:
+	- there is only one colour used in the image
+	- the first pixel of the screen capture is black
+	"""
+	"""
+	Iterating over a large bitmap image in python is slow (~4s depending on machine, number of pixels).
+	Due to no native support for calculating the number of black pixels, NVDA's screenBitmap module is avoided.
+	wxWidgets has native support for calculating the usage of colours within an image.
+	Using this method to calculate a colour usage histogram in native code takes ~0.4s comparably.
+
+	An example of iterating over a screen bitmap, using NVDA's screenBitmap module.
+	from screenBitmap import ScreenBitmap
+	from itertools import chain
+	from winGDI import RGBQUAD
+	screenCapture = ScreenBitmap(screenSize[0], screenSize[1]).captureImage(0, 0, screenSize[0], screenSize[1])
+	def _isPixelBlack(pixel: RGBQUAD) -> bool:
+		return pixel.rgbRed == 0 and pixel.rgbGreen == 0 and pixel.rgbBlue == 0
+	numNonBlackPixels = len(list(filter(lambda p: not _isPixelBlack(p), chain.from_iterable(screenCapture))))
+	return numNonBlackPixels == 0
+	"""
+	screen = wx.ScreenDC()
+	screenSize = screen.GetSize()
+	bmp: wx.Bitmap = wx.Bitmap(screenSize[0], screenSize[1])
+	mem = wx.MemoryDC(bmp)
+	mem.Blit(0, 0, screenSize[0], screenSize[1], screen, 0, 0)  # copy screen over to bmp
+	del mem  # Release bitmap
+	img: wx.Image = bmp.ConvertToImage()

```

Comment:
@feerrenrut - at what stage is a buffer grab fail a concern?

Would a check like the following be an improvement?

```suggestion
	bmp: wx.Bitmap = wx.Bitmap(screenSize[0], screenSize[1])
	setCanaryBits(bmp)
	mem = wx.MemoryDC(bmp)
	assert areCanaryBitsSet(mem)
	mem.Blit(0, 0, screenSize[0], screenSize[1], screen, 0, 0)  # copy screen over to bmp
	assert not areCanaryBitsSet(mem)
	del mem  # Release bitmap
	img: wx.Image = bmp.ConvertToImage()
```

Reply:
Note: This doesn't handle if `screen` is initialised wrong - which is handled by `wx` and we have little control over.